src/webview_provider/insightsPanel.ts

```diff
@@ -0,0 +1,267 @@
+import { commands, TextEditor, window, workspace } from "vscode";
+import { getProjectRelativePath, provideSingleton } from "../utils";
+import { DBTProjectContainer } from "../manifest/dbtProjectContainer";
+import { DBTProject } from "../manifest/dbtProject";
+import { TelemetryService } from "../telemetry";
+import {
+  AltimateWebviewProvider,
+  HandleCommandProps,
+  UpdateConfigProps,
+} from "./altimateWebviewProvider";
+import { AltimateRequest } from "../altimate";
+import { SharedStateService } from "../services/SharedStateService";
+
+type UpdateConfigPropsOrArray = UpdateConfigProps | UpdateConfigProps[];
+interface DeferConfig {
+  deferToProduction: boolean;
+  favorState: boolean;
+  manifestPathForDeferral: string;
+}
+@provideSingleton(InsightsPanel)
+export class InsightsPanel extends AltimateWebviewProvider {
+  public static readonly viewType = "dbtPowerUser.Insights";
+  protected viewPath = "/insights";
+  protected panelDescription = "Toggle Defer to prod and other features";
+
+  public constructor(
+    dbtProjectContainer: DBTProjectContainer,
+    protected altimateRequest: AltimateRequest,
+    telemetry: TelemetryService,
+    protected emitterService: SharedStateService,
+  ) {
+    super(dbtProjectContainer, altimateRequest, telemetry, emitterService);
+
+    window.onDidChangeActiveTextEditor(
+      async (event: TextEditor | undefined) => {
+        if (event === undefined) {
+          return;
+        }
+
+        if (this._panel) {
+          const currentProjectRoot = await this.getCurrentProjectRoot();
+          const currentConfig: Record<string, DeferConfig> = await workspace
+            .getConfiguration("dbt")
+            .get("deferConfigPerProject", {});
+
+          this._panel!.webview.postMessage({
+            command: "updateDeferConfig",
+            args: currentConfig[currentProjectRoot] || {},
+          });
+        }
+      },
+    );
+  }
+
+  private async getCurrentProjectRoot() {
+    const currentFilePath = window.activeTextEditor?.document.uri;
+    if (!currentFilePath) {
+      throw new Error("Invalid current file");
+    }
+
+    const currentProject =
+      this.dbtProjectContainer.findDBTProject(currentFilePath);
+    if (!currentProject?.projectRoot) {
+      throw new Error("Invalid current project root");
+    }
+
+    return getProjectRelativePath(currentProject.projectRoot);
+  }
+  private async updateDeferConfig(
+    syncRequestId: string | undefined,
+    params: UpdateConfigPropsOrArray,
+  ) {
+    try {
+      console.log("Updating defer config", params);
+      // defer config is preview feature, then check keys
+      if (!this.altimateRequest.handlePreviewFeatures()) {
+        throw new Error("Invalid credentials");
+      }
+
+      if (Object.keys(params).every((key) => /^\d+$/.test(key))) {
+        params = Object.values(params);
+      }
+
+      let updateConfigs: UpdateConfigProps[];
+      if (Array.isArray(params)) {
```

Comment:
instead of doing this, it will be better to restrict only to array types so we dont need to different cases